spec/lib/os_places_api/client_spec.rb

```diff
@@ -255,6 +255,16 @@
       expect(Postcode.where(postcode: postcode).pluck(:results)).to eq([os_places_api_results])
     end
 
+    it "should update the `updated_at` property even if no changes were made" do
+      updated_at = 1.day.ago
+      Postcode.create(postcode: postcode, results: successful_response[:results], updated_at: updated_at)
+      stub_request(:get, api_endpoint)
+        .to_return(status: 200, body: successful_response.to_json)
+
+      client.update_postcode(postcode)
+      expect(Postcode.find_by(postcode: postcode).updated_at.strftime("%A")).to eq(Time.now.strftime("%A"))
```

Comment:
Is %A the right thing to use here? That's capitalised name of day, is it not?

Reply:
You're correct 👍 I did initially try just doing `.updated_at).not_to eq(updated_at)` but the test failed. Digging into it, there were microseconds of discrepancy between the `updated_at` variable and the `updated_at` field in the database:

```
$ Postcode.find_by(postcode: postcode).updated_at
Sun, 05 Jun 2022 10:51:54.729208000 BST +01:00
$ updated_at
Sun, 05 Jun 2022 10:51:54.729208376 BST +01:00
```

I'm surprised by the discrepancy, but figuring out exactly why it's there doesn't feel important. My test really only cares that the `updated_at` field is updated, so parsing the day from the `ActiveSupport::TimeWithZone` instance and checking that it is now today instead of yesterday seemed sensible. There are lots of ways I could have done that, but I picked the `%A` option somewhat arbitrarily.

(Also could have accomplished everything with listeners, e.g. `allow(Postcode).to_receive(:find_by).and_return(mock); expect(mock).to_receive(:touch)`. But it feels less brittle to test the actual behaviour).

Reply:
Okay, that makes sense!